app/src/schemas.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Zod schemas for validating server responses
+ */
+
+import { z } from "zod";
+
+export const UUIDSchema = z.uuid({ version: "v4" });
+
+// Response from /api/v1/token
+export const TokenResponseSchema = z.object({
+  expiration: z.string(),
+  token: z.string().min(1),
+  journalist_uuid: UUIDSchema,
+  journalist_first_name: z.nullable(z.string()),
+  journalist_last_name: z.nullable(z.string()),
+});
+
+export const SourceMetadataSchema = z.object({
```

Comment:
if we want to allow + keep unknown keys, we could use [looseobject](https://zod.dev/api?id=zlooseobject)?

but I think the zod default behavior of dropping unknown keys feels the most correct to me: the client won't error, but it'll have a different "version" and continually request from the server, until the client updates to also know about a new field.